subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestingIntegrationTest.groovy

```diff
@@ -78,33 +78,30 @@ class TestingIntegrationTest extends JUnitMultiVersionIntegrationSpec {
         """
 
         when:
-        executer.withArgument("-Dtest.debug")
+        executer.withArgument("-Dtest.debug").expectDeprecationWarning()
+        then:
+        succeeds("test")
+        output.contains("System property 'test.debug' has been deprecated and is scheduled to be removed in Gradle 5.0. Use --debug-jvm to enable remote debugging of tests.")
 
+        when:
+        executer.withArgument("-D:test.debug").expectDeprecationWarning()
         then:
         succeeds("test")
+        output.contains("System property ':test.debug' has been deprecated and is scheduled to be removed in Gradle 5.0. Use --debug-jvm to enable remote debugging of tests.")
+
+        expect:
+        succeeds("test", "--debug-jvm")
     }
 
     def "configures test task when test.single property is set"() {
         given:
         buildFile << """
-            import org.gradle.api.internal.tasks.properties.PropertyVisitor
-            import org.gradle.api.internal.tasks.properties.PropertyWalker
-            import org.gradle.api.internal.tasks.TaskInputFilePropertySpec
-            import org.gradle.api.internal.tasks.TaskPropertyUtils
-
             apply plugin: 'java'
 
             task validate() {
                 doFirst {
+                    assert test.candidateClassFiles.empty
                     assert test.includes  == ['**/pattern*.class'] as Set
-                    boolean hasSourceFiles = false
```

Comment:
❓I assume this change is related to the addition of `@SkipWhenEmpty` annotation. I'm not seeing what this test is actually trying to do and how the annotation helps this case.

Reply:
Yes.  I'm not sure why it was so complicated.

If you look at https://github.com/gradle/gradle/blob/57e6a6d22335150a366cfd46a73a747cd4417ca3/subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestTaskIntegrationTest.groovy#L48 without `@SkipWhenEmpty`, this test fails.  With it, it passes.

Just thinking about this... there may be a hole:

- I have `src/test/java/MyTest.java`
- I run `./gradlew test -Dtest.single=MyTst` (typo)
- I think `:test` would be shown as "skipped" vs failing with "No tests found"

I don't think we have a test for that.

Reply:
I added a test for this https://github.com/gradle/gradle/pull/5235/files#diff-e3007116e49ab5a0e15c42e438cbbef6R172